Assets/Tile walking mini game/Scripts/GridManager.cs

```diff
@@ -11,18 +11,26 @@ public class GridManager : MonoBehaviour
 
     [SerializeField] private TileColorManager tileColorManager;
 
-    [SerializeField] private ObstaclesPosition[][] obstacleLayouts;
+    private ObstaclesPosition[][] obstacleLayouts;
 
     [SerializeField] private GameObject TilePrefab;
 
-    public Action WinEvent;
+    public Action<GameState> GameEventUpdatedEvent;
 
-    public Action LoseEvent;
-    public int UntouchedTileCount { get; private set; }
+    public Action<Tile> TileStatusChangedEvent;
+
+    public int touchedTiles_Percent { get; private set; } = 0;
 
     public int GridWidth { get; private set; } = 5;
     public int GridHeight { get; private set; } = 5;
 
+    [SerializeField] private GameState currentGameState;
+
+    [Range(20, 95)]
+    [SerializeField] private int partialWinThreshold_Percent = 75;

```

Comment:
I dont really like the underscore in the name as its not really a convention but not a big deal